xray/plotting/plotting.py

```diff
@@ -221,6 +276,23 @@ def _plot2d(plotfunc):
         if None, use the default for the matplotlib function
     add_colorbar : Boolean, optional
         Adds colorbar to axis
+    vmin, vmax : floats, optional
+        Values to anchor the colormap, otherwise they are inferred from the
+        data and other keyword arguments. When a diverging dataset is inferred,
+        one of these values may be ignored.
+    cmap : matplotlib colormap name or object, optional
+        The mapping from data values to color space. If not provided, this
+        will be either be ``viridis`` (if the function infers a sequential
+        dataset) or ``RdBu_r`` (if the function infers a diverging dataset).
+    center : float, optional
+        The value at which to center the colormap. Passing this value implies
+        use of a diverging colormap.
+    robust : bool, optional
+        If True and ``vmin`` or ``vmax`` are absent, the colormap range is
+        computed with robust quantiles instead of the extreme values.
+    extend : {'neither', 'both', 'min', 'max'}, optional
```

Comment:
If you look at how extend is inferred, if the entire range of plotted values falls inside vmin and vmax, it won't show extended colorbars.

On Wed, Jul 29, 2015 at 1:15 PM, Joe Hamman <notifications@github.com>
wrote:

>> @@ -221,6 +276,23 @@ def _plot2d(plotfunc):
>>          if None, use the default for the matplotlib function
>>      add_colorbar : Boolean, optional
>>          Adds colorbar to axis
>> +    vmin, vmax : floats, optional
>> +        Values to anchor the colormap, otherwise they are inferred from the
>> +        data and other keyword arguments. When a diverging dataset is inferred,
>> +        one of these values may be ignored.
>> +    cmap : matplotlib colormap name or object, optional
>> +        The mapping from data values to color space. If not provided, this
>> +        will be either be ``viridis`` (if the function infers a sequential
>> +        dataset) or ``RdBu_r`` (if the function infers a diverging dataset).
>> +    center : float, optional
>> +        The value at which to center the colormap. Passing this value implies
>> +        use of a diverging colormap.
>> +    robust : bool, optional
>> +        If True and ``vmin`` or ``vmax`` are absent, the colormap range is
>> +        computed with robust quantiles instead of the extreme values.
>> +    extend : {'neither', 'both', 'min', 'max'}, optional
> You definitely want to keep the `extend` keyword argument available. Imagine the case where you are plotting a statistic with range of [-1, 1].  If you set `vmin` and `vmax` to -1 and 1 respectively, and don't have control of the colorbar extend option, you will lead your viewers to believe the statistic's range goes beyond the range [-1, 1].
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/xray/xray/pull/499/files#r35806217